### PR TITLE
feat(Resources/Grpah): (Anomaly Detection-2) change the display of the enveloppe size to be an actual colored area

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
Add threshold to display enveloppe size 

screen (Mocked result):
ADdETAILS

### CODE REVIEW

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { equals, prop, isNil } from 'ramda';
+import { Threshold } from '@visx/threshold';
+import { curveBasis } from '@visx/curve';
+import { ScaleLinear, ScaleTime } from 'd3-scale';
+
+import { TimeValue } from '../models';
+import { getTime } from '../timeSeries';
+
+interface Props {
+  getYScale;
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx]: Add type if possible

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  graphHeight: number;
+  leftScale: ScaleLinear<number, number>;
+  regularLines;
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx]: Same comment here

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  rightScale: ScaleLinear<number, number>;
+  secondUnit: string;
+  thirdUnit: string;
+  timeSeries: Array<TimeValue>;
+  xScale: ScaleTime<number, number>;
+}
+
+const ThresholdAD = ({
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx]:
```suggestion
const AnomalyDetectionEnvelopeThreshold = ({
```

Does it make any sense to add some memoization here? cc @Thebarda

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    invert: invertY0,
+    leftScale,
+    rightScale,
+    secondUnit,
+    unit: unitY0,
+  });
+
+  const X = (timeValue): number => xScale(getTime(timeValue)) as number;
+  const Y1 = (timeValue): number => y1Scale(prop(metricY1, timeValue)) ?? null;
+  const Y0 = (timeValue): number => y0Scale(prop(metricY0, timeValue)) ?? null;
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx]: Should we lower case these variable names?

File: www/front_src/src/Resources/Graph/Performance/Lines/RegularLine.tsx [hunk 5]
```diff
@@ -36,13 +36,21 @@ const RegularLine = ({
   transparency,
   graphHeight,
 }: Props): JSX.Element => {
+  const strokeWidth =
+    equals(metric, 'connection_lower_thresholds') ||
+    equals(metric, 'connection_upper_thresholds')
+      ? 0.1
+      : 0.8;
+
+  const isHighlight = highlight ? 2 : strokeWidth;
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Lines/RegularLine.tsx]:
```suggestion
  const isHighlighted = highlight ? 2 : strokeWidth;
```

File: www/front_src/src/Resources/Graph/Performance/Lines/RegularLine.tsx [hunk 6]
```diff
@@ -36,13 +36,21 @@ const RegularLine = ({
...
+
   const props = {
     curve: Curve.curveLinear,
     data: timeSeries,
     defined: (value): boolean => !isNil(value[metric]),
     opacity: highlight === false ? 0.3 : 1,
     stroke: lineColor,
-    strokeWidth: highlight ? 2 : 1,
+    strokeWidth: isHighlight,
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Lines/RegularLine.tsx]:
```suggestion
    strokeWidth: isHighlighted,
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 7]
```diff
@@ -100,6 +105,24 @@ const Lines = ({
 
   const regularLines = difference(lines, stackedLines);
 
+  const isLegendClicked = lines?.length <= 1;
+
+  const isDisplayedThreshold =
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
  const isThresholdDisplayed =
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 8]
```diff
@@ -100,6 +105,24 @@ const Lines = ({
...
+    equals(type, ResourceType.anomalydetection) && !isLegendClicked;
+
+  const propsThresholdAD = {
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
  const thresholdProps = {
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 9]
```diff
@@ -119,6 +142,7 @@ const Lines = ({
         yScale={stackedYScale}
       />
       <g>
+        {isDisplayedThreshold && <ThresholdAD {...propsThresholdAD} />}
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
        {isThresholdDisplayed && <ThresholdAD {...thresholdProps} />}
```

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx (hunk 3 above)
Comment (Thebarda) [reply www/front_src/src/Resources/Graph/Performance/AnomalyDetection/ThresholdAD.tsx]: As it is a heavy component, it makes sense to add memoization on non-function props

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionEnvelopeThreshold.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    invert: invertY0,
+    leftScale,
+    rightScale,
+    secondUnit,
+    unit: unitY0,
+  });
+
+  const x = (timeValue): number => xScale(getTime(timeValue)) as number;
+  const y1 = (timeValue): number => y1Scale(prop(metricY1, timeValue)) ?? null;
+  const y0 = (timeValue): number => y0Scale(prop(metricY0, timeValue)) ?? null;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionEnvelopeThreshold.tsx]:
```suggestion
  const getXPoint = (timeValue): number => xScale(getTime(timeValue)) as number;
  const getY1Point = (timeValue): number => y1Scale(prop(metricY1, timeValue)) ?? null;
  const getY0Point = (timeValue): number => y0Scale(prop(metricY0, timeValue)) ?? null;
```

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionEnvelopeThreshold.tsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        fillOpacity: 0.1,
+      }}
+      clipAboveTo={0}
+      clipBelowTo={graphHeight}
+      curve={curveBasis}
+      data={timeSeries}
+      id={`${y0.toString()}${y1.toString()}`}
+      x={x}
+      y0={y0}
+      y1={y1}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionEnvelopeThreshold.tsx]:
```suggestion
      x={getXPoint}
      y0={getY0Point}
      y1={getY1Point}
```

File: www/front_src/src/Resources/Graph/Performance/Graph/index.tsx [hunk 12]
```diff
@@ -490,6 +501,27 @@ const GraphContent = ({
 
   const commentTitle = isCommentPermitted ? '' : t(labelActionNotPermitted);
 
+  const stackedLines = getSortedStackedLines(lines);
+
+  const regularLines = difference(lines, stackedLines);
+
+  const isLegendClicked = lines?.length <= 1;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Graph/index.tsx]:
```suggestion
  const isLegendClicked = lte(length(lines), 1);
```

File: www/front_src/src/Resources/Graph/Performance/Graph/index.tsx [hunk 13]
```diff
@@ -526,6 +558,11 @@ const GraphContent = ({
               xScale={xScale}
             />
             <MemoizedLines
+              AnomalyDetectionEnvelope={
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Graph/index.tsx]:
```suggestion
              anomalyDetectionEnvelope={
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 14]
```diff
@@ -1,26 +1,29 @@
...
   getYScale,
 } from '../timeSeries';
 
-import RegularLine from './RegularLine';
 import RegularAnchorPoint from './AnchorPoint/RegularAnchorPoint';
+import RegularLine from './RegularLine';
 import StackedLines from './StackedLines';
 
 interface Props {
+  AnomalyDetectionEnvelope: ReactNode;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
  anomalyDetectionEnvelope: ReactNode;
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 15]
```diff
@@ -79,6 +82,7 @@ const Lines = ({
   graphHeight,
   timeTick,
   displayTimeValues,
+  AnomalyDetectionEnvelope,
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
  anomalyDetectionEnvelope,
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx [hunk 16]
```diff
@@ -119,6 +123,7 @@ const Lines = ({
         yScale={stackedYScale}
       />
       <g>
+        {AnomalyDetectionEnvelope}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]:
```suggestion
        {anomalyDetectionEnvelope}
```

File: www/front_src/src/Resources/Graph/Performance/Lines/index.tsx (hunk 14 above)
Comment (Thebarda) [reply www/front_src/src/Resources/Graph/Performance/Lines/index.tsx]: This prop should be optional

**Pull request closed without merging**